src/app/pcasts/dao/episodes_dao.py

```diff
@@ -46,8 +46,15 @@ def is_recommended_by_user(episode_id, user_id):
     .first()
   return optional_recommendation is not None
 
-def search_episode(search_name, offset, max_search):
-  possible_episodes = Episode.query.filter \
-      (Episode.title.like(search_name+'%')) \
-      .offset(offset).limit(max_search).all()
-  return possible_episodes
+def search_episode(search_name, offset, max_search, user_id):
+  possible_episode_ids = [
+      tup[0] for tup in
+      Episode.query.\
+      with_entities(Episode.id).\
```

Comment:
Hmm I'm a bit confused as to what `with_entities` is used for, I read the documentation but I still don't really understand tbh

Reply:
It's the equivalent of a SQL projection.